alcs-api/src/application/application.controller.ts

```diff
@@ -53,22 +57,30 @@ export class ApplicationController {
     return {
       ...mappedApplication[0],
       statusDetails: application.status,
+      typeDetails: application.type,
     };
   }
 
   @Post()
   @UserRoles(...ANY_AUTH_ROLE)
-  async add(@Body() application: ApplicationDto): Promise<ApplicationDto> {
-    const entity = await this.mapToEntity(application);
-    const app = await this.applicationService.createOrUpdate(entity);
+  async create(
+    @Body() application: CreateApplicationDto,
+  ): Promise<ApplicationDto> {
+    const type = await this.applicationTypeService.get(application.type);
+    const app = await this.applicationService.createOrUpdate({
+      ...application,
+      type,
+      typeUuid: type.uuid,
+      title: 'Do we still need title?',
+    });
     const mappedApps = await this.mapApplicationsToDtos([app]);
     return mappedApps[0];
   }
 
   @Patch()
   @UserRoles(...ANY_AUTH_ROLE)
   async update(
-    @Body() application: ApplicationPartialDto,
+    @Body() application: ApplicationUpdateDto,
   ): Promise<ApplicationDto> {
     const existingApplication = await this.applicationService.get(
       application.fileNumber,
```

Comment:
I wonder if it worth adding the application number into ServiceValidationException:

`throw new ServiceValidationException(`File not found ${application.fileNumber}`);`

Reply:
Good suggestion, added!